### ISSUE TITLE
Allow external istiod and config cluster system namespaces to be different

### ISSUE THREAD

**Issue by frankbu** [istio/istio#32147]
**Describe the feature request**

When using an [external control plane configuration](https://preliminary.istio.io/latest/docs/setup/additional-setup/external-controlplane/), the external istiod and config cluster system namespaces must be the same. In the documented example they are both using the ns `external-istiod`. One can extend the example to deploy a second external istiod on the same external cluster (for a different mesh) in another ns, for example `external-istiod-2`. In that case, the second mesh's config cluster will also need to use system namespace `external-istiod-2`.

It would be much nicer if the two namespaces could be different, allowing the mesh cluster to be configured in a standard way, independent of the number of external istiods that may be running on the external cluster. In fact, ideally, the system namespace on the config cluster could be `istio-system`, by default, no different than for a local istiod mesh.

**Describe alternatives you've considered**

[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Additional context**

**Comment by howardjohn** [comment]
Is this a docs issue or is there a technical constraint?

**Comment by frankbu** [comment]
@howardjohn There seem to be some places in the code where istiod uses its root namespace, assuming it's the same as the root namespace on the config cluster. So, we probably need some enhancement to allow configuring both namespaces for the external istiod and make sure the correct one is always used.

**Comment by howardjohn** [comment]
When do we ever use the root namespace of the config cluster? I thought we get root namespace from mesh config only

**Comment by frankbu** [comment]
Seems the first problem we observed is here it is using istiod's ns to create the rootCA: https://github.com/istio/istio/blob/2a69579d6db5c6fc8f6983b4977393ead751aeef/pilot/pkg/bootstrap/istio_ca.go#L377. Shouldn't it be using the meshconfig root namespace, instead of `opts.Namespace` here? Don't know if there are other similar issues or not.

**Comment by howardjohn** [comment]
Root namespace and the system namespace are not the same ( they are
actually recommended to be different, although we don't follow that...).
Root namespace refers to the namespace that configs with inheritance apply
globally.

**Comment by frankbu** [comment]
I updated the title and description after off-line discussion with @howardjohn, where I realized that I was using "root namespace" in the description, instead of "system namespace", which was confusing the issue.

To clarify further, the root namespace is where inherited config (e.g., DestinationRules) go. It shouldn't necessarily be he same as the system namespace (usually istio-system) where istiod, default gateways, configmaps, webhooks, etc. go. The problem is that we don't want the system namespace to be the same on the data plane cluster as on the external cluster where istiod is deployed. For example, on a multi-tenanted external cluster istiod is deployed in a namespace like: abc123defxxx-external-istiod, but we want the configmaps, webhooks, etc., on the data plane (config) cluster to be in a simple namespace like istio-system.

In summary, we need to configure istiod to be running in `namespace1` on the control plane cluster, but have it access and work with system resources (configmaps, etc.) in `namespace2` on the config cluster.

**Comment by frankbu** [comment]
This looks like more work to implement than it's worth. Closing.
